HpToolsLauncher/MtbxManager.cs

```diff
@@ -1,260 +1,275 @@
+                }
+            }
+
+            List<TestInfo> retval = new List<TestInfo>();
+            XDocument doc = XDocument.Parse(xmlContent);
+
+            XmlSchemaSet schemas = new XmlSchemaSet();
+
+            var assembly = Assembly.GetExecutingAssembly();
+
+            var schemaStream = assembly.GetManifestResourceStream("HpToolsLauncher.MtbxSchema.xsd");
+
+            XmlSchema schema = XmlSchema.Read(schemaStream, null);
+
+            schemas.Add(schema);
+
+            string validationMessages = string.Empty;
+            doc.Validate(schemas, (o, e) =>
+            {
+                validationMessages += e.Message + Environment.NewLine;
+                ConsoleWriter.ErrorSummaryLines.Add(e.Message);
+            });
+
+            if (!string.IsNullOrWhiteSpace(validationMessages))
+            {
+                ConsoleWriter.WriteLine("mtbx schema validation errors: " + validationMessages);
+            }
+            try
+            {
+                var root = doc.Root;
+                foreach (var test in GetElements(root, "Test"))
+                {
+                    string path = GetAttribute(test, "path").Value;
+                    if (string.IsNullOrWhiteSpace(path))
+                    {
+                        string err = string.Format(Resources.EmptyPathAttributeValue, path);
+                        ConsoleWriter.WriteLine($"Error: {err}");
+                        ConsoleWriter.ErrorSummaryLines.Add(err);
+                        Launcher.ExitCode = Launcher.ExitCodeEnum.Failed;
+                        continue;
+                    }
+                    if (!Directory.Exists(path))
+                    {
+                        string err = string.Format(Resources.GeneralFileNotFound, path);
+                        ConsoleWriter.WriteLine($"Error: {err}");
+                        ConsoleWriter.ErrorSummaryLines.Add(err);
+                        Launcher.ExitCode = Launcher.ExitCodeEnum.Failed;
+                        continue;
+                    }
+
+                    // optional report path attribute (report base directory, for backward compatibility)
+                    XAttribute xReportBasePath = GetAttribute(test, "reportPath");
+                    string reportBasePath = null;
+                    if (xReportBasePath != null)
+                    {
+                        reportBasePath = xReportBasePath.Value;
+                    }
+
+                    // optional report directory path attribute (exact report path)
+                    XAttribute xReportExactPath = GetAttribute(test, "reportExactPath");
+                    string reportExactPath = null;
+                    if (xReportExactPath != null)
+                    {
+                        reportExactPath = xReportExactPath.Value;
+                    }
+
+                    XAttribute xname = GetAttribute(test, "name");

```

Comment:
Is that possible `GetAttribute` method may return `null`? If so, here may cause crash

Reply:
Unfortunately yes, good catch.